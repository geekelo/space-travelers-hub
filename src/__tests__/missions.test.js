import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import Missions from '../components/missions';

const mockStore = configureMockStore();
const initialState = {
  missions: {
    missions: [
      {
        key: '1',
        missionId: '1',
        mission: 'Today rescue',
        description: 'Interesting',
        reserved: true,
      },
      // ... Add more rockets as needed for testing
    ],
    status: 'done',
    error: false,
  },
};
const store = mockStore(initialState);

describe('Rocket Component', () => {
  it('matches the snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );

    // Take a snapshot of the rendered output
    expect(container).toMatchSnapshot();
  });
});
