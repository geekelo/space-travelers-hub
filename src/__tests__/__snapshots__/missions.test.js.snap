// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rocket Component matches the snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <table
      class="table table table-striped table-bordered table-hover"
    >
      <thead
        class="thead"
      >
        <tr>
          <td>
            Mission
          </td>
          <td>
            Description
          </td>
          <td>
            Status
          </td>
          <td>
             
          </td>
        </tr>
      </thead>
      <tbody>
        <tr
          class="missionList"
        >
          <td
            class="mission_name"
          />
          <td>
            Interesting
          </td>
          <td
            class="membership_stat"
          >
            <button
              class="active_member"
              type="button"
            >
              Active member
            </button>
          </td>
          <td
            class="reservation_stat"
          >
            <button
              class="leave_mission"
              type="button"
            >
              Leave Mission
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
